cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/PeerState.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.tethering;
+
+import java.util.Objects;
+
+/**
+ * Information about tethered peers that's returned by REST APIs.
+ */
+public class PeerState extends PeerBase {
+  private final TetheringConnectionStatus connectionStatus;
+
+  public PeerState(String name, String endpoint, TetheringStatus tetheringStatus, PeerMetadata metadata,
+                   TetheringConnectionStatus connectionStatus) {
+    super(name, endpoint, tetheringStatus, metadata);
+    this.connectionStatus = connectionStatus;
+  }
+
+  public TetheringConnectionStatus getConnectionStatus() {
```

Comment:
Will the status be more than active/inactive? If it is always being boolean, probably just need to have a `isActive` or `isAlive` method instead?

Reply:
changed to `isActive`